### PR TITLE
Cleaned up styles, external view functions, and added an interface

### PR DESCRIPTION
**Pull request by joaoritter**
Several things in this PR:

Small things:
1. Consolidated many references to Money pools to just 'mp'.
2. Variables declared within a function start with an underscore.
3. Any time the word Money pool shows up in comments, it's now "Money pool" as opposed to "MoneyPool" or "MP" or any other variation.

Big things:
1. External views fit for clients are written. Tests were modified to make use of these.
2. The contract now adheres to an interface, which will make contract inter-dependency cleaner and capable of handling upgrades without breaking.
3. the `sustain` function takes in a `beneficiary` parameter. This allows a msg.sender to make a payment on another address's behalf. This may be useful when a contract is taking a fee through Fountain. See the new examples/ contract for an example.

Notes:
Before this PR, I tried a branch where MoneyPools were their own contract. I took this idea to its conclusion (see the branch), but i decided to abandon it because it would be costly to spin up new MPs during normal execution of the contract. The current solution works best I think.

### CODE REVIEW

File: contracts/FountainV1.sol [hunk 1]
```diff
@@ -1,691 +1,759 @@
...
-    /// @dev Returns a copy of the given MoneyPool with reset sustainments.
-    /// @param moneyPoolId The id of the MoneyPool to base the new MoneyPool on.
-    /// @param start The start date to use for the new MoneyPool.
-    /// @return newMoneyPoolId The new MoneyPool ID.
-    function _createMoneyPoolFromId(uint256 moneyPoolId, uint256 start)
-        private
-        returns (uint256 newMoneyPoolId)
-    {
-        MoneyPool storage currentMoneyPool = moneyPools[moneyPoolId];
+    /// @dev Take the amount that should be redistributed to the given sustainer by the given owner's Money pools.
```
Comment (tauren) [inline contracts/FountainV1.sol:599]: This description is confusing

File: contracts/FountainV1.sol [hunk 2]
```diff
@@ -1,691 +1,759 @@
...
     {
-        // Use the old end if the current time is still within the duration.
-        if (oldEnd.add(duration) > now) return oldEnd;
-        // Otherwise, use the closest multiple of the duration from the old end.
-        uint256 distanceToStart = (now.sub(oldEnd)).mod(duration);
-        return now.sub(distanceToStart);
+        MoneyPool storage _mp = mps[_mpId];
+
+        // Return 0 if there's no surplus.
+        if (!_mp.exists || _mp.target >= _mp.total) return 0;
```
Comment (tauren) [inline contracts/FountainV1.sol:662]: Small nit... maybe reverse logic to `_mp.total < _mp.target`

**Review by tauren — APPROVED** [review]
This is looking great. I went through it carefully and am not seeing any significant issues. Just a couple nits, but looks like we can land this.

File: contracts/examples/YourProject.sol [hunk 3]
```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.6.0 <0.8.0;
+
+import "../interfaces/IFountainV1.sol";
+
+/// @dev This contract is an example of how you can use Fountain to fund your own project.
+contract YourProject {
```
[the remark this answers is not in the document]
Comment (joaoritter) [reply contracts/examples/YourProject.sol:7]: i like the thought. lets keep this in mind

File: contracts/FountainV1.sol [hunk 4]
```diff
@@ -1,691 +1,759 @@
...
     }
 
-    /// @notice A message sender can collect funds that have been used to sustain it's MoneyPools.
-    /// @param amount The amount to collect.
+    /// @dev A message sender can collect specific funds that have been used to sustain it's Money pools.
+    /// @param _mpId The ID of the Money pool to tap.
+    /// @param _amount The amount to tap.
     /// @return success If the collecting was a success.
-    function collectSustainments(uint256 amount)
+    function collectSustainments(uint256 _mpId, uint256 _amount)
```
[the remark this answers is not in the document]
Comment (joaoritter) [reply contracts/FountainV1.sol:374]: i was curious what you'd think. I removed tapped in favor of collect

File: contracts/FountainV1.sol (hunk 1 above)
Comment (joaoritter) [reply contracts/FountainV1.sol:599]: did a take2

**Pull request merged**